modules/otel-collector-components.adoc

```diff
@@ -869,7 +938,47 @@ This extension supports traces, metrics, and logs.
 <10> The path to the client's private key file that is used with the client certificate if needed for authentication.
 <11> Sets a timeout for the token client's request.
 <12> You can assign the authenticator configuration to an OTLP exporter.
+[id="filestorage-extension_{context}"]
+=== Filestorage extension
+
+The File Storage extension can persist state to the local file system.
+The extension requires read and write access to a directory. A default directory can be used, but it must already exist in order for the extension to operate.
+
+The OAuth2Client extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocol.
+This extension supports traces, metrics, and logs.
+
+.OpenTelemetry Collector custom resource with configured FileStorage extension to persist as OTLP sendig queue.
+[source,yaml]
+----
+  config: |
+    extensions:
+      file_storage/all_settings:
+        directory: /var/lib/otelcol/mydir # <1>
+        timeout: 1s # <2>
+        compaction:
+          on_start: true # <3>
+          directory: /tmp/ # <4>
+          max_transaction_size: 65_536 # <5>
+        fsync: false # <6>
 
+    exporters:
+      otlp:
+        sending_queue:
+          storage: file_storage/all_settings # <7>
```

Comment:
Should we add a callout for `<7>` below? Or is this just a dangling typo?

Reply:
@frzifus, :slightly_smiling_face:  what do you mean by "Register" here? Could you please explain it to me in some other words?

Reply:
To be able to use the `file_storage/all_settinghs` extension here, you need to create an instance or start it internally.  Thats done by listing it in the `servcie::extensions` section. Does this make sense?

Reply:
@frzifus, what do you mean by "buffered on disk" in your suggestion below?

`7: Register the file_storage/all_settings component in the otlp exporter that should be buffered on disk.`

Reply:
Usually the process will keep all measurements in an in-memory queue. If the process restarts, that data will be lost. Using the `file_storage` extension this data will be written to disk. If the collector restarts, the measurements will still exist and can still be transmitted to a remote backend.

Reply:
@frzifus, :sweat_smile: what is "disk"? I know this sounds like a very dumb question, but specifically? Is it a Persistent Volume?

Reply:
yes :+1: